tools/Google.Cloud.Tools.VersionCompat.Tests/StructTest.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Google.Cloud.Tools.VersionCompat.Tests.Struct
+{
+    namespace NoChange.A { public struct S { } }
+    namespace NoChange.B { public struct S { } }
+
+    public interface I1 { }
+    internal interface I2 { }
+
+    namespace InterfaceAdded.A { public struct S { } }
+    namespace InterfaceAdded.B { public struct S : I1 { } }
+
+    namespace InterfaceRemoved.A { public struct S : I1 { } }
+    namespace InterfaceRemoved.B { public struct S { } }
+
+    namespace NonExportedInterfaceAdded.A { public struct S { } }
+    namespace NonExportedInterfaceAdded.B { public struct S : I2 { } }
+
+    namespace NonExportedInterfaceRemoved.A { public struct S : I2 { } }
+    namespace NonExportedInterfaceRemoved.B { public struct S { } }
+
+    namespace GenericConstraintAdded1.A { public struct S<A> { } }
```

Comment:
Should we have tests for `where A : unmanaged`? New in C# 7 IIRC. Can't see us using it, admittedly.

Reply:
I guess so! TODO added in `ClassStructInterface.cs`.